Add description about telemetry of builtin functions usage (#6028)
pingcap/docs: telemetry.md

- Deployment characteristics, such as the size of hardware (CPU, memory, disk), TiDB components versions, OS name.
- The status of query requests in the system, such as the number of query requests and the duration.
- Component usage, for example, whether the Async Commit feature is in use or not.

> - Builtin functions usage information, including the count of functions used in SQL queries.

Reviewer: ```suggestion
- Usage information of builtin functions, including the average number of functions used per SQL query.
```

```suggestion
- Usage information of builtin functions, including the count of functions used in SQL queries.
```

Author: Actually it is a count per function. 

I'm not sure how to explain it clearly. What do you think about this?

Reviewer: Would this be ok?
```suggestion
- Usage information of builtin functions, including per-function statistics on how often the functon gets called. The arguments to the functions are not included.
```

Author: Nice, I'll modify the part in Chinese version according to this as well.

> - Usage information of builtin functions, including per-function statistics on how often the functon gets called. The arguments to the functions are not included.

Reviewer 2: ```suggestion
- The usage information of built-in functions, including per-function statistics on how often the function gets called. The arguments for the functions are not included.
```
